Source/WebCore/layout/integration/inline/InlineIteratorTextBox.cpp

```diff
@@ -41,10 +41,29 @@ TextBoxIterator TextBox::nextTextBox() const
 
 LayoutRect TextBox::selectionRect(unsigned rangeStart, unsigned rangeEnd) const
 {
+    bool isCaretCase = rangeStart == rangeEnd;
+    rangeEnd = std::min(end(), rangeEnd);
```

Comment:
It would be good to understand why this is needed. Can it be handled by clamp()?

Reply:
I also think we don't since we are clamping anyway. I've removed it and adjusted the checks for caret within box. It still passes our Range tests.